test/integration/transaction.test.ts

```diff
@@ -0,0 +1,155 @@
+import { describe, before } from 'mocha'
+import { Express } from 'express'
+import { expect } from 'chai'
+
+import createHttpServer from '../../src/server'
+import { get } from '../helper/routeHelper'
+import {
+  seed,
+  cleanup,
+  seededCapacityId,
+  nonExistentId,
+  seededTransactionId4,
+  exampleDate,
+} from '../seeds'
+import { TransactionState, TransactionApiType, TransactionType } from '../../src/models/transaction'
+
+describe('transaction', () => {
+  let app: Express
+
+  before(async function () {
+    app = await createHttpServer()
+  })
+
+  beforeEach(async function () {
+    await seed()
+  })
+
+  afterEach(async function () {
+    await cleanup()
+  })
+
+  // handling all errors - error handling reflects whats in code, 422 first because being handled by TSOA/OPEN-API
+  describe('if transaction id is not UUID or bad request', () => {
+    it('returns 422 along with validation error', async () => {
+      const { status, body } = await get(app, '/transaction/123-not-uuid')
+
+      expect(status).to.equal(422)
+      expect(body).to.deep.contain({
+        fields: {
+          transactionId: {
+            message: "Not match in '[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[89ABab][0-9A-Fa-f]{3}-[0-9A-Fa-f]{12}'",
+            value: '123-not-uuid'
+          }
+        },
+        name: 'ValidateError',
+        message: 'Validation failed'
+      })
+    })
+  })
+
+  describe('when requested transaction can not be found', () => {
+    it('returns 404 along with the message', async () => {
+      const { status, body } = await get(app, `/transaction/${nonExistentId}`)
+
+      expect(status).to.equal(404)
+      expect(body).to.equal('transaction not found')
+    })
+  })
+
+  it('returns empty array if database contains 0 transactions', async () => {
+    cleanup()
+    const { status, body } = await get(app, '/transaction')
+
+    expect(status).to.equal(200)
+    expect(body).to.be.an('array').that.is.empty
+  })
+
+
+  it('gets transaction by id', async () => {
+    const { status, body }= await get(app, `/transaction/${seededTransactionId4}`)
+
+    expect(status).to.equal(200)
+    expect(body).to.deep.contain({
+        id: seededTransactionId4,
+        apiType: TransactionApiType.order,
+        transactionType: TransactionType.creation,
+        localId: seededCapacityId,
+        state: TransactionState.submitted,
+        submittedAt: exampleDate,
+        updatedAt: exampleDate,
+      })
+  })
+
+  // TODO assert for limit (when set on the db)
+  it('returns all transactions', async () => {
+    const { status, body } = await get(app, '/transaction')
+
+    expect(status).to.equal(200)
+    expect(body.length).to.equal(5)
+    expect(body).to.deep.include.members([
```

Comment:
could change to `expect length > 0` and only check the the first transaction. Otherwise every time we change one of the seeded transactions (which get used for testing various different routes) we have to update this test.  We do have a ticket to use test fixtures instead though so will be resolved after that https://digicatapult.atlassian.net/browse/L3-100 and could ignore this

Reply:
well, i would debate that and argue that we should trust our seeds and if we retrieve all we should assert what we get not pick one. Due to their differences e.g. types, statuses and etc.

As a solution we could bring this number down or other solution for heavy comparison i use something like **snapshotting** - auto generated. so you do not need to updated manually 10 lines, you generate a new snapshot when you update properties or entities or...